app/assets/javascripts/components/locator-maps.js

```diff
@@ -0,0 +1,154 @@
+var mymap;
+var myLocationMarker;
+var spinner;
+
+(function(window, $) {
+  'use strict';
+
+  var $mapid = $('#mapid');
+
+  // Only run the following on pages which have a map
+  if (!$mapid.length)
+    return;
+
+  var lat = $mapid.data('originLat');
+  var lng = $mapid.data('originLng');
+
+  mymap = L.map('mapid').setView([lat, lng], 4);
+
+  L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+    attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="http://mapbox.com">Mapbox</a>',
+    maxZoom: 18
+  }).addTo(mymap);
+
+  L.Control.Spinner = L.Control.extend({
+    onAdd: function(map) {
+        var img = L.DomUtil.create('img');
+
+        img.src = $mapid.data('spinnerPath');
+        img.style.width = '200px';
+
+        return img;
+    },
+
+    onRemove: function(map) {
+        // Nothing to do here
+    }
+  });
+
+  L.control.spinner = function(opts) {
+    return new L.Control.Spinner(opts);
+  }
+
+  spinner = L.control.spinner({ position: 'bottomleft' });
+
+  var myIcon = L.icon({
+      iconUrl: $mapid.data('markerIconPath'),
+      iconRetinaUrl: $mapid.data('markerIcon2xPath'),
+      iconAnchor: [17, 48],
+      popupAnchor: [0, -48],
+      shadowUrl: $mapid.data('markerShadowPath'),
+      shadowAnchor: [13, 41]
+  });
+
+  // remove existing markers
+  for (var marker of markers) {
+    marker.remove();
+  }
+  markers = [];
+
+  // place current markers with links to the corresponding results, and store them for later possible removal
+  for (var loc of locations) {
+    var marker = L.marker([loc['lat'], loc['lng']], {icon: myIcon, title: loc['name'], riseOnHover: true}).addTo(mymap);
```

Comment:
For readability's sake, throw in some breaks like:

```js
var marker = L.marker([loc['lat'], loc['lng']], {
  icon: myIcon, 
  title: loc['name'], 
  riseOnHover: true
}).addTo(mymap);
```

For referencing keys in an object, you can use dot notation instead of `loc['lat']`:

- `loc.name`
- `loc.lat`
- `loc.lng`